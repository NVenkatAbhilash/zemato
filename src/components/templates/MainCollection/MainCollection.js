import React from "react";
import { makeStyles } from "@material-ui/core/styles";
import { Grid } from "@material-ui/core";
import CollectionViewItem from "../../molecules/CollectionViewItem/CollectionViewItem";

const useStyles = makeStyles({
    root:{
        padding: "0 40px 40px 40px",
    }
});

const MainCollection = ({foodImg, ...props}) => {
  const classes = useStyles();

  return (
    <Grid container justify="center" className={classes.root}>
      <Grid item sm={12}>
        <CollectionViewItem  foodImg={foodImg} {...props}/>
      </Grid>
    </Grid>
  );
};

export default MainCollection;
