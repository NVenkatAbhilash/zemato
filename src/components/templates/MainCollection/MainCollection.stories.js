import React from "react";
import MainCollection from "./MainCollection";
import foodPhoto from "../../../resources/images/collection2.jpg";

export default {
    title: "template/MainCollection",
    component: MainCollection,
}


const Template = ({...args})=>(
    <MainCollection {...args} />
);


export const Primary = Template.bind({});
Primary.args = {
    foodImg: foodPhoto,
}

