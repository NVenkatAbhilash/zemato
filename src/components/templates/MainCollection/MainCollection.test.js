import React from "react";
import ReactDOM from "react-dom";
import { BrowserRouter } from "react-router-dom";
import "@testing-library/jest-dom";
import MainCollection from "./MainCollection";
import renderer from "react-test-renderer";

it("renders MainCollection snapshot correctly", () => {
  const tree = renderer
    .create(
      <BrowserRouter>
        <MainCollection />
      </BrowserRouter>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it("renders MainCollection without crashing", () => {
  const div = document.createElement("div");
  ReactDOM.render(
    <BrowserRouter>
      <MainCollection />
    </BrowserRouter>,
    div
  );
});
