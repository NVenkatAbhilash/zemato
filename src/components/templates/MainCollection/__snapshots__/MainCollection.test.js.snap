// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders MainCollection snapshot correctly 1`] = `
<div
  className="MuiGrid-root makeStyles-root-1 MuiGrid-container MuiGrid-justify-xs-center"
>
  <div
    className="MuiGrid-root MuiGrid-item MuiGrid-grid-sm-12"
  >
    <div
      className="MuiGrid-root makeStyles-root-2 MuiGrid-container MuiGrid-align-items-xs-flex-end"
      data-testid="main"
      style={
        Object {
          "backgroundImage": "linear-gradient(rgba(0,0,0, 0.1), rgb(0, 0, 0, .6)), url(undefined)",
        }
      }
    >
      <div
        className="MuiGrid-root makeStyles-content-4 MuiGrid-item"
      >
        <h6
          className="MuiTypography-root MuiTypography-h6"
          data-testid="special"
        />
        <h6
          className="MuiTypography-root MuiTypography-subtitle2"
          data-testid="place"
        >
           Places
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root makeStyles-arrowIcon-5 MuiSvgIcon-fontSizeSmall"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M10 17l5-5-5-5v10z"
            />
          </svg>
        </h6>
      </div>
    </div>
  </div>
</div>
`;
